src/resolvers/dockerfileTemplate.ts

```diff
@@ -18,8 +32,13 @@ export class DockerfileTemplateResolver implements Resolver {
 		this.hasDockerfileTemplate = true;
 	}
 
-	public needsEntry(filepath: string) {
-		return basename(filepath) === 'Dockerfile.template';
+	public needsEntry(entryPath: ParsedPathPlus, specifiedDockerfilePath?: string) {
+		return (
+			entryPath.ext === '.template' &&
+			(specifiedDockerfilePath
+				? entryPath.unparsed === specifiedDockerfilePath
+				: entryPath.minusExt === 'Dockerfile')
```

Comment:
Two cases:
* If a `specifiedDockeriflePath` was specified, say `'service/MyDockerfile.template'`, then this resolver (`DockerfileTemplateResolver `) will only match a tar entry whose full path is identical to that (`entryPath.unparsed === specifiedDockerfilePath`), and provided that the `specifiedDockeriflePath` has a `'.template'` file extension.
* Otherwise, it will match `Dockerfile.template` at the root of the project directory tree, as `entryPath.minusExt` is the full entry path minus the file extension, which must be exactly `'Dockerfile'`. This root-only rule was previously enforced in `resolveTarStreamOnEntry()` through a test: `if (path.normalize(name).split(path.sep).length === 1)`. That test had been added by a recent-ish pull request:
_fix: Correctly resolve contexts with multiple bundle descriptions_
https://github.com/balena-io-modules/resin-bundle-resolve/pull/32/files

Note that both `entryPath` and `specifiedDockerfilePath` are normalized through `TarUtils.normalizeTarEntry()` before the call to `needsEntry()`, so they won't have leading or trailing slashes or redundant path components.